source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -41,28 +41,17 @@ namespace
                + " ms";
     }
 
-    std::string makeIrisList(const std::vector<Zivid::Settings> &settingsVector)
+    std::string makeSettingList(const std::vector<Zivid::Settings> &settingsVector,
+                             std::function<std::string(Zivid::Settings)> setting)
     {
-        std::string irisList = "{ ";
+        std::string settingList = "{ ";
         for(size_t i = 0; i < settingsVector.size(); i++)
         {
-            irisList += settingsVector.at(i).iris().toString();
-            if(i + 1 != settingsVector.size()) irisList += ", ";
+            settingList += setting(settingsVector.at(i));
```

Comment:
This was advansed stuff :) But if I understand what you are trying to achieve this is not the right tool. Templates is the tool for this job:

```
template<typename Target>
std::string makeSettingList(const std::vector<Zivid::Settings> &settingsVector)
{
  std::string settingList = "{ ";
  for(for const auto& setting : settingsVector)
  {
      settingList += setting.get<Target>().toString();
      if(i + 1 != settingsVector.size()) settingList += ", ";
  }
  settingList += " }";
  return settingList;
}

// Then use it like this

std::cout << "  exposure time = " << makeSettingList<Zivid::Settings::ExposureTime>(settingsVector) << std::endl;

// Old version:

std::cout << "  exposure time = " << makeSettingList(settingsVector, [](const Zivid::Settings &s){
								       return s.exposureTime().toString();
								     }) << std::endl;
```

Reply:
Done, like this (need the index)
`template<typename Target>
    std::string makeSettingList(const std::vector<Zivid::Settings> &settingsVector)
    {
        std::string settingList = "{ ";
        for(size_t i = 0; i < settingsVector.size(); i++)
        {
            settingList += settingsVector.at(i).get<Target>().toString();
            if(i + 1 != settingsVector.size()) settingList += ", ";
        }
        settingList += " }";
        return settingList;
    }`